src/scirpy/ir_dist/metrics.py

```diff
@@ -392,6 +394,698 @@ def _compute_block(self, seqs1, seqs2, origin):
         return result
 
 
+class TCRdistDistanceCalculator:
+    """Computes pairwise distances between TCR CDR3 sequences based on the "tcrdist" distance metric.
+
+    The code of this class is heavily based on: https://github.com/agartland/pwseqdist/blob/master/pwseqdist
+
+    Using default weight, gap penalty, ntrim and ctrim is equivalent to the
+    original distance published in Dash et al, (2017).
+
+    Parameters
+    ----------
+    dist_weight : int
+        Weight applied to the mismatch distances before summing with the gap penalties
+    gap_penalty : int
+        Distance penalty for the difference in the length of the two sequences
+    ntrim/ctrim : int
+        Positions trimmed off the N-terminus (0) and C-terminus (L-1) ends of the peptide sequence. These symbols will be ignored
+        in the distance calculation.
+    fixed_gappos : bool
+        If True, insert gaps at a fixed position after the cysteine residue statring the CDR3 (typically position 6).
+        If False, find the "optimal" position for inserting the gaps to make up the difference in length
+    cutoff:
+        Will eleminate distances > cutoff to make efficient
+        use of sparse matrices.
+    n_jobs:
+        Number of jobs (processes) to use for the pairwise distance calculation
+    """
+
+    def __init__(
+        self,
+        dist_weight=3,
+        gap_penalty=4,
+        ntrim=3,
+        ctrim=2,
+        fixed_gappos=True,
+        cutoff: Union[None, int] = None,
+        n_jobs: Union[None, int] = None,
+    ):
+        if cutoff is None:
+            cutoff = 20
+        if n_jobs is None:
+            n_jobs = 1
+
+        self.dist_weight = dist_weight
+        self.gap_penalty = gap_penalty
+        self.ntrim = ntrim
+        self.ctrim = ctrim
+        self.fixed_gappos = fixed_gappos
+        self.cutoff = cutoff
+        self.n_jobs = n_jobs
+
+    parasail_aa_alphabet = "ARNDCQEGHILKMFPSTWYVBZX"
+    parasail_aa_alphabet_with_unknown = "ARNDCQEGHILKMFPSTWYVBZX*"
+    tcr_dict_distance_matrix = {
```

Comment:
In such a case you can add the magic comments
```python
# fmt: off
<your code> 
# fmt: on
```

to disable autoformatting for the given section.